app/models/doi.rb

```diff
@@ -675,6 +675,24 @@ def self.find_by_id(id)
     )
   end
 
+  def self.stats_query(query, options={})
+    aggregations = {created: { date_histogram: { field: 'created', interval: 'year', format: 'year', order: { _key: "desc" }, min_doc_count: 1 },
+                    aggs: { bucket_truncate: { bucket_sort: { size: 10 } } } },
+                   }
+
+    from = 0
+    sort = [{ created: "asc", uid: "asc" }]
+
+    __elasticsearch__.search({
+      size: options.dig(:page, :size),
```

Comment:
You don't need from and sort, so I would leave them out. I would hardwire `size: 0`, as you are only interested in the aggregations.

I would write `query` so that the function also supports stats for a single provider or client.

The aggregation should have a `size` larger than 10, so that we include all DataCite DOIs. We currently start at 2010 (when DataCite was founded), so 11 years.

Have you looked at `track_total_hits`? Not sure what the best advice is for your query. https://www.elastic.co/guide/en/elasticsearch/reference/master/search-request-body.html#request-body-search-track-total-hits

Reply:
@mfenner, could you please have a look at the query that I'm constructing. It is not quite working but I haven't been able to figure out why yet.